### PR TITLE
Add answer links to the knowledge check section of Ruby/Nested Collections

### LINKED ISSUE #22920: Add missing Knowledge Checks, Learning Outcomes, and Knowledge Check links to Ruby/Nested Collections
Related to #22467

This task is to go through the lesson referenced in the title of this issue and (as applicable):

- Add missing Learning Outcomes
- Add missing Knowledge Checks
- Add answer links in the Knowledge Checks

If the lesson already has Learning Outcomes or Knowledge Checks they do not need to be added to. Many lessons only need answer links added.

Knowledge check links should consist of an `<a>` tag which adds the `knowledge-check-link` class and links to the part of the lesson (or the external resource that is linked in the lesson) where the answer to the Knowledge Check can be found. Knowledge Check links should **not** point to external resources that weren't linked to as part of the lesson content or assignment. You can link to a specific line in the lesson like so:

```html
Lorem ipsum dolor sit amet, <span id='example-knowledge-check'>consectetur adipiscing elit</span>, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to the page where the span with this id lives -->
<a class='knowledge-check-link' href='#example-knowledge-check'>How do you return a value from a function?</a>
```
Or you can link to just a section of the lesson by just linking to the section heading and following [these](https://github.com/TheOdinProject/curriculum/wiki/Contributing-to-Wiki#anchor-linking) rules for determining how to format the link. Example:

```html
### Lorem ipsum dolor sit amet

sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to a heading on the page-->
<a class='knowledge-check-link' href='#lorem-ipsum-dolor-sit-amet'>How do you return a value from a function?</a>
```

### Acceptance Criteria

- [ ] Lesson referenced in the title has Learning Outcomes, Knowledge Checks, and Knowledge Check links
- [ ] Knowledge Check links all link to the proper place in the lesson, or the proper external resource that was linked to earlier in the lesson
- [ ] Knowledge Check styling should **not** change

Please do not start work on this unless you've been assigned to it. Comment on this issue and ping me @wise-king-sullyman if you would like to be assigned to it.

Additionally, we would like for contributors to only take on one of these issues at a time. Please only request assignment for (and begin work on) an issue after the Learning Outcome/Knowledge Check/Knowledge Check link issue you're currently assigned to has been approved and merged.

### PR DESCRIPTION
**Pull request by JoshDevHub**
- [x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [x] The PR title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 
#### 1.Describe the changes made and include why they are necessary or important:

I added answer links to each of the items in the knowledge check section. I split a few of the checks up because it would've been impossible to link to the correct place in the lesson with a single link. An example is the current top item in the knowledge-check list asks `What is a nested array and hash?` but nested arrays and nested hashes are introduced in very different sections of the page. I also combined the `What are nested arrays/hashes` with `What data is held by nested arrays/hashes` because their answers are adjacent on the page and can be conveniently linked together.

#### 2. If this PR is related to an open issue please reference it with the `#` operator and the issue number (and any [relevant keywords](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests)) below:

closes #22920

### CODE REVIEW

**Comment by wise-king-sullyman** [comment]
@jmsmith1018 in the future can you mention the lesson being changed (when only one lesson is being changed) in the PR title? Makes things easier for us to keep an eye on things at a glance.

I already updated the title here, so just something to know on any PRs you do going forward :slightly_smiling_face:

File: ruby_programming/basic_ruby/nested_collections.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ teacher_mailboxes[0][4]
 #=> nil
 ~~~
 
-If you want a nil value returned when trying to access an index of a nonexistent nested element, you can use the `#dig` method. This method can also be used when accessing a nonexistent index inside of an existing nested element.
+<span id='dig-method-knowledge-check'>If you want a nil value returned when trying to access an index of a nonexistent nested element, you can use the `#dig` method.<span> This method can also be used when accessing a nonexistent index inside of an existing nested element.
```
Comment (wise-king-sullyman) [inline ruby_programming/basic_ruby/nested_collections.md:70]:
```suggestion
<span id='dig-method'>If you want a nil value returned when trying to access an index of a nonexistent nested element, you can use the `#dig` method.<span> This method can also be used when accessing a nonexistent index inside of an existing nested element.
```

File: ruby_programming/basic_ruby/nested_collections.md [hunk 2]
```diff
@@ -333,11 +333,15 @@ This section contains helpful links to other content. It isn't required, so cons
...
-- Why is the `#dig` method useful?
-- How do you add data to a nested array and hash?
-- How do you delete data in a nested array and hash?
-- How do you create a new nested array that is not mutable?
-- How do you iterate over a nested array and hash?
+- <a class='knowledge-check-link' href='#nested-arrays'>What is a nested array? What data is useful to store in a nested array?</a>
+- <a class='knowledge-check-link' href='#nested-hashes'>What is a nested hash? What data is useful to store in a nested hash?</a>
+- <a class='knowledge-check-link' href='#accessing-elements'>How do you access elements in a nested array?</a>
+- <a class='knowledge-check-link' href='#accessing-data'>How do you access data in a nested hash?</a>
+- <a class='knowledge-check-link' href='#dig-method-knowledge-check'>Why is the `#dig` method useful?<a>
```
Comment (wise-king-sullyman) [inline ruby_programming/basic_ruby/nested_collections.md:340]:
```suggestion
- <a class='knowledge-check-link' href='#dig-method'>Why is the `#dig` method useful?<a>
```

File: ruby_programming/basic_ruby/nested_collections.md [hunk 3]
```diff
@@ -333,11 +333,15 @@ This section contains helpful links to other content. It isn't required, so cons
...
+- <a class='knowledge-check-link' href='#adding-and-removing-elements'>How do you add elements to a nested array?</a>
+- <a class='knowledge-check-link' href='#adding-and-removing-data'>How do you add data to a nested hash?</a>
+- <a class='knowledge-check-link' href='#remove-elements-nested-array'>How do you delete elements from a nested array?</a>
+- <a class='knowledge-check-link' href='#deleting-data-nested-hash'>How do you delete data in a nested hash?</a>
+- <a class='knowledge-check-link' href='#create-immutable-nested-arrays'>How do you create a new nested array that is not mutable?</a>
+- <a class='knowledge-check-link' href='#iterating-over-nested-a-array'>How do you iterate over a nested array?</a>
```
Comment (wise-king-sullyman) [inline ruby_programming/basic_ruby/nested_collections.md:346]:
```suggestion
- <a class='knowledge-check-link' href='#iterating-over-a-nested-array'>How do you iterate over a nested array?</a>
```

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
This looks great overall! I have a few suggestions/change requests then it should be good to go!

**Comment by JoshDevHub** [comment]
Committed the changes. I'll also be sure to include the lesson name in the PR next time. Thanks for the feedback :+1:

**Review by wise-king-sullyman — APPROVED** [review]
Good catches on those missing `/`s!

This looks great, thank you for your contribution :smile:

**Pull request merged**